operator/controller/webhook/introute/sync.py

```diff
@@ -76,16 +79,64 @@ def get_mounts(self) -> List[Mapping]:
         return volume_mounts
 
 
-def _spring_cloud_k8s_config(parent) -> Optional[Mapping]:
+def _get_updated_props_srcs(related_configmaps, props_srcs) -> List:
+    """Replaces each props_src name with the latest ConfigMap based on creationTimestamp and prop_src base name"""
+    updated_props = []
+
+    if not props_srcs or not related_configmaps:
+        _LOGGER.debug(
+            f"Either props_srcs or configmaps were missing\n Props Sources: {props_srcs} \nConfigmaps: {related_configmaps}"
+        )
+        return updated_props
+
+    for prop_src in props_srcs:
+        matching_configmaps = {}
+        for cm_name, cm in related_configmaps.items():
+            if "name" in prop_src and prop_src["name"] in cm_name:
+                matching_configmaps[cm_name] = cm
+            elif "labels" in prop_src:
+                labels = prop_src["labels"]
+                for label in labels:
+                    if (
+                        label in cm["metadata"]["labels"].keys()
+                        and labels[label] == cm["metadata"]["labels"][label]
+                    ):
+                        matching_configmaps[cm_name] = cm
+        if not matching_configmaps:
+            # No updates
+            continue
+
+        # Get the most recent Configmap
+        updated_cm = max(
+            matching_configmaps.values(),
+            key=lambda cm: datetime.strptime(
+                cm["metadata"]["creationTimestamp"], CM_DATETIME_FORMAT
+            ),
+        )
+
+        _LOGGER.debug(f"Updated ConfigMap: {updated_cm}")
+        updated_prop = {"name": updated_cm["metadata"]["name"]}
+        if updated_prop not in updated_props:
+            updated_props.append(updated_prop)
+
+    _LOGGER.info(f"Updated Props: {updated_props}")
+    return updated_props
+
+
+def _spring_cloud_k8s_config(observed) -> Optional[Mapping]:
     """Generates the spring-cloud-kubernetes config that's passed as an env var to the Spring app"""
+    parent = observed["parent"]
     metadata = parent["metadata"]
-
     props_srcs = parent["spec"].get("propSources")
     secret_srcs = parent["spec"].get("secretSources")
+    _LOGGER.info(observed)
```

Comment:
Might be a duplicate since the top-level `sync` function already logs the `observed` object.

Reply:
I think I meant to log the `related` field when I was debugging. I'll remove it.